[java] Add 1.5

adriens opened the pull request:
# ❔  About

☕ Java 1.5 (aka. J2SE 5.0) and older are missing

# 🎫 Related issue(s)

- https://github.com/endoflife-date/endoflife.date/issues/2474

adriens committed:
Add ☕ Java 1.5 (aka. J2SE 5.0)

cf #2474

BiNZGi commented on products/java.md:
```diff
@@ -125,6 +125,15 @@ releases:
+    lts: false
+    support: 2015-04-01
+    eol: 2009-11-01
+    latest: "1_5_0_22"
```
Latest from
https://github.com/endoflife-date/release-data/blob/b197fef895f13fc9fce100325a7d0c16314a98c6/releases/java.json#L161
> "5.0u85": "2015-04-14",

Also here: https://www.java.com/releases/ -> search for `5.0u85`

adriens replied:
Done

BiNZGi commented on products/java.md:
```diff
@@ -125,6 +125,15 @@ releases:
+    eol: 2009-11-01
+    latest: "1_5_0_22"
+    releaseDate: 2004-09-29
+    link: https://www.oracle.com/java/technologies/javase/releasenotes-v150.html
```
Better we use this `link`: https://www.oracle.com/java/technologies/javase/advancedv5-support-relnotes.html
From here: https://www.java.com/releases/ -> search for 5.0u85

adriens replied:
Done.

BiNZGi commented on products/java.md:
```diff
@@ -125,6 +125,15 @@ releases:
+-   releaseCycle: "5"
+    lts: false
+    support: 2015-04-01
+    eol: 2009-11-01
```
```suggestion
    support: 2009-11-03
    eol: 2009-11-03
```
`support` cannot be later as `eol`.
source: https://www.java.com/en/download/help/java_5_eol.html
> The Java version 5.0 reached the End Of Service Life (EOSL) on November 3, 2009

adriens committed:
Update products/java.md

Thanks for the fix ;-p

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

adriens committed:
Better link  for latest Java SE 5

cf https://github.com/endoflife-date/endoflife.date/pull/2475#discussion_r1100128597

BiNZGi commented on products/java.md:
```diff
@@ -125,6 +125,15 @@ releases:
+    lts: false
+    support: 2009-11-03
+    eol: 2009-11-03
+    latest: "1.5.0_85"
```
```suggestion
    latest: "5.0u85"
```
In https://github.com/endoflife-date/endoflife.date/pull/2475#discussion_r1100126757 I thought changing like that.

adriens committed:
Update products/java.md

LOL, I was asking myself the same question ;-p

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

BiNZGi reviewed (approved):
LGTM now
